src/java/org/apache/cassandra/index/sai/disk/format/IndexDescriptor.java

```diff
@@ -0,0 +1,530 @@
+    }
+
+    public SearchableIndex newSearchableIndex(SSTableContext sstableContext, IndexContext indexContext)
+    {
+        return version.onDiskFormat().newSearchableIndex(sstableContext, indexContext);
+    }
+
+    public PerSSTableWriter newPerSSTableWriter() throws IOException
+    {
+        return version.onDiskFormat().newPerSSTableWriter(this);
+    }
+
+    public PerIndexWriter newPerIndexWriter(StorageAttachedIndex index,
+                                            LifecycleNewTracker tracker,
+                                            RowMapping rowMapping)
+    {
+        return version.onDiskFormat().newPerIndexWriter(index, this, tracker, rowMapping);
+    }
+
+    public boolean isPerSSTableBuildComplete()
+    {
+        return version.onDiskFormat().isPerSSTableBuildComplete(this);
+    }
+
+    public boolean isPerIndexBuildComplete(IndexContext indexContext)
+    {
+        return version.onDiskFormat().isPerIndexBuildComplete(this, indexContext);
+    }
+
+    public boolean isIndexEmpty(IndexContext indexContext)
+    {
+        return isPerIndexBuildComplete(indexContext) && numberOfComponents(indexContext) == 1;
+    }
+
+    public long sizeOnDiskOfPerIndexComponents(IndexContext indexContext)
+    {
+        registerPerIndexComponents(indexContext);
+        if (perIndexComponents.containsKey(indexContext.getIndexName()))
+            return perIndexComponents.get(indexContext.getIndexName())
+                                     .stream()
+                                     .map(c -> Pair.create(c, indexContext.getIndexName()))
+                                     .map(onDiskPerIndexFileMap::get)
+                                     .filter(java.util.Objects::nonNull)
+                                     .filter(File::exists)
+                                     .mapToLong(File::length)
+                                     .sum();
+        return 0;
+    }
+
+    @VisibleForTesting
+    public long sizeOnDiskOfPerIndexComponent(IndexComponent indexComponent, IndexContext indexContext)
+    {
+        if (perIndexComponents.containsKey(indexContext.getIndexName()))
+            return perIndexComponents.get(indexContext.getIndexName())
+                                     .stream()
+                                     .filter(c -> c == indexComponent)
+                                     .map(c -> Pair.create(c, indexContext.getIndexName()))
+                                     .map(onDiskPerIndexFileMap::get)
+                                     .filter(java.util.Objects::nonNull)
+                                     .filter(File::exists)
+                                     .mapToLong(File::length)
+                                     .sum();
+        return 0;
+    }
+
+    public boolean validatePerIndexComponents(IndexContext indexContext)
+    {
+        logger.info("validatePerIndexComponents called for " + indexContext.getIndexName());
```

Comment:
```suggestion
        logger.info("Validating per-column index components for " + indexContext.getIndexName());
```
(Goes back to the argument around how we use "per-column" and "per-index" generally. OCD, but we might as well find something we like.)

Reply:
I think that `per-column index` is the correct terminology. Btw, I've removed the `for index name` from the log message because this should have been using `indexContext.logMessage` that adds the index name in a uniform manner.